substrate/frame/revive/src/precompiles/builtin/system.rs

```diff
@@ -31,6 +31,12 @@ sol! {
 	interface ISystem {
 		/// Computes the BLAKE2 256-bit hash on the given input.
 		function hashBlake256(bytes memory input) external pure returns (bytes32 digest);
+		/// Retrieve the account id for a specified `H160` address.
+		///
+		/// # Note
+		///
+		/// If no mapping exists for `addr`, the fallback account id will be returned.
+		function toAccountId(address input) external pure returns (bytes32 account_id);
```

Comment:
This potentially reads state.

```suggestion
		function toAccountId(address input) external view returns (bytes32 account_id);
```